ext/web/06_streams.js

```diff
@@ -4755,41 +4804,18 @@ const readableStreamAsyncIteratorPrototype = ObjectSetPrototypeOf({
 
       /** @type {Deferred<IteratorResult<any>>} */
       const promise = new Deferred();
-      /** @type {ReadRequest} */
-      const readRequest = {
-        chunkSteps(chunk) {
-          promise.resolve({ value: chunk, done: false });
-        },
-        closeSteps() {
-          readableStreamDefaultReaderRelease(reader);
-          promise.resolve({ value: undefined, done: true });
-        },
-        errorSteps(e) {
-          readableStreamDefaultReaderRelease(reader);
-          promise.reject(e);
-        },
-      };
+      const readRequest = new ReadableStreamAsyncIteratorReadRequest(
+        reader,
+        promise,
+      );
 
       readableStreamDefaultReaderRead(reader, readRequest);
-      return PromisePrototypeThen(promise.promise, (result) => {
-        reader[_iteratorNext] = null;
-        if (result.done === true) {
-          reader[_iteratorFinished] = true;
-          return { value: undefined, done: true };
-        }
-        return result;
-      }, (reason) => {
-        reader[_iteratorNext] = null;
-        reader[_iteratorFinished] = true;
-        throw reason;
-      });
```

Comment:
Where did this code go?

Reply:
here https://github.com/denoland/deno/blob/6369ad452b6520a8d4de2778bdfc69c72e9a9df5/ext/web/06_streams.js#L4766-L4782

Reply:
Can you add a comment for future readers so they know where to change things if the spec changes? Thanks!

Reply:
Sorry @lucacasonato I was OOO, just read this.

AFAIK these state variables are not part of the spec, but needed to pass WPT.

Reply:
They are defined in this spec: https://webidl.spec.whatwg.org/#es-default-asynchronous-iterator-object

They handle cases like ensuring that calling `next()` after it has returned done won't throw

Reply:
> They are defined in this spec: https://webidl.spec.whatwg.org/#es-default-asynchronous-iterator-object

Thanks, wasn't aware of that.

Added a comment @lucacasonato, feel free to edit it.